CMakeLists.txt

```diff
@@ -16,179 +16,106 @@
 
-if(UNIX OR MSYS)
-	find_program(LDCONFIG_EXECUTABLE "ldconfig")
-endif()
+# Link available libraries
+target_link_libraries(libdpp PRIVATE OpenSSL::SSL)
+target_link_libraries(libdpp PRIVATE Threads::Threads)
+target_link_libraries(libdpp PRIVATE ZLIB::ZLIB)
+
+# Determine if voice support will be enabled
+if(Sodium_FOUND AND Opus_FOUND)
+	target_link_libraries(libdpp PRIVATE sodium)
+	target_link_libraries(libdpp PRIVATE opus)
 
-if(MSVC)
-	if(CMAKE_BUILD_TYPE STREQUAL "Debug")
-		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /std:c++17 /Od /DEBUG /Zi /W4 /sdl /MP")
-	else()
-		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /std:c++17 /O2 /Oi /Oy /GL /Gy /W4 /sdl /MP")
-	endif()
+	message(STATUS "Detected ${Green}libsodium${ColourReset} and ${Green}libopus${ColourReset}. VOICE support will be ${Green}enabled${ColourReset}")
+	target_compile_definitions(libdpp PRIVATE HAVE_VOICE)
 else()
-	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -pthread -g -O3 -fPIC -rdynamic")
+	message(STATUS "Could not detect ${Green}libsodium${ColourReset} and/or ${Green}libopus${ColourReset}. VOICE support will be ${Red}disabled${ColourReset}")
 endif()
 
-set(modules_dir "src")
-
-file(GLOB subdirlist ${modules_dir}/dpp)
-
-foreach (fullmodname ${subdirlist})
-    get_filename_component(modname ${fullmodname} NAME)
-    set (modsrc "")
-	
-	file(GLOB modsrc "${modules_dir}/dpp/*.cpp" "${modules_dir}/dpp/events/*.cpp")
-	
-	if(BUILD_SHARED_LIBS)
-        add_library(${modname} SHARED ${modsrc})
-	else()
-		add_library(${modname} STATIC ${modsrc})
-	endif()
-
-	if (WIN32)
-		target_link_libraries(${modname} PUBLIC "${VCPKG_INSTALLATION_64_BITS_FOLDER}/lib/libssl.lib"
-												"${VCPKG_INSTALLATION_64_BITS_FOLDER}/lib/libcrypto.lib"
-												"${VCPKG_INSTALLATION_64_BITS_FOLDER}/lib/zlib.lib")
-	else()
-		target_link_libraries(${modname} PUBLIC ${OPENSSL_CRYPTO_LIBRARY} ${OPENSSL_SSL_LIBRARY} ${ZLIB_LIBRARIES})
-	endif()
-
-	if (HAVE_VOICE)
-		target_link_libraries(${modname} PUBLIC ${sodium_LIBRARY_RELEASE} ${OPUS_LIBRARIES})
-		
-		include_directories(${OPUS_INCLUDE_DIRS} ${sodium_INCLUDE_DIR})
-	endif()
-endforeach()
-
-target_compile_features(dpp PRIVATE cxx_std_17)
-
-target_link_libraries(test PUBLIC ${modname})
-
-if(WIN32)
-	configure_file("${VCPKG_INSTALLATION_64_BITS_FOLDER}/bin/zlib1.dll" "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}" COPYONLY)
+# Build tests
+if(DPP_BUILD_TESTS)
+	add_executable(test src/test/test.cpp)
+	target_link_libraries(test PRIVATE libdpp)
 endif()
 
-if(NOT WIN32)
-	# Installation
-
-	include(GNUInstallDirs)
-	install(TARGETS dpp LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
-	install(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
-	install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -DLDCONFIG_EXECUTABLE=${LDCONFIG_EXECUTABLE} -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/PostInstall.cmake)")
-endif()
+return()
+###################################################################################
```

Comment:
Not sure why all this is commented out, will it be put back in after the whole PR is done? :) Needed to install the lib and run ldconfig

Reply:
I commented it out, so I could slowly rewrite the CMake as 3.0 without those lines interfering. I won't be added back. Once CPack is completed, it would handle the rest of what those lines do.